### PR TITLE
Created sharpen function

### PR DESCRIPTION
**Pull request by mani-kohli**
sharpen function with blurring image integrated

### CODE REVIEW

File: picfixPy/sharpen.py [hunk 1]
```diff
@@ -1,20 +1,112 @@
...
+        
+        
+    # Guassian blurring of image
+    
+    # approximation of Gaussian blur 3x3 kernel
+    # kernel attribution: https://en.wikipedia.org/wiki/Kernel_(image_processing)
+    kernel = np.array([[1.0,2.0,1.0], [2.0,4.0,2.0], [1.0,2.0,1.0]])
+    kernel = kernel / np.sum(kernel)
+    
+    # attribution: https://bit.ly/2EbS4XD
```
Comment (GeorgeJJW) [inline picfixPy/sharpen.py:62]: It's important that we give credits where credits are due, good job attributing the author of this blurring approach!

File: picfixPy/sharpen.py [hunk 2]
```diff
@@ -1,20 +1,112 @@
...
+    
+    # https://en.wikipedia.org/wiki/Unsharp_masking
+    # https://homepages.inf.ed.ac.uk/rbf/HIPR2/unsharp.htm
+    
+    # blurred image 
+    blurred = image_array_sum
+    
+    # concept: sharpened = original + (original − blurred) × amount.
+    # attribution: https://bit.ly/2V3FAr0
+    sharpened = float(intensity + 1) * input_img - float(intensity) * blurred
```
Comment (GeorgeJJW) [inline picfixPy/sharpen.py:86]: So this step is essentially expanding `(original − blurred) × amount` to `original * amount - blurred * amount`, okay, this should give us the enhanced edges, I'm following along :grinning:. Don't we have to add the original image back at some point?

File: picfixPy/sharpen.py [hunk 3]
```diff
@@ -1,20 +1,112 @@
...
+    sharpened = np.maximum(sharpened, np.zeros(sharpened.shape))
+    sharpened = np.minimum(sharpened, 255 * np.ones(sharpened.shape))
+    sharpened = sharpened.round().astype(np.uint8)
```
Comment (GeorgeJJW) [inline picfixPy/sharpen.py:89]: Nice vectorized way of restricting the value of the pixel back to between 0 and 255, and rounding it back to an integer. Nicely done.

File: picfixPy/sharpen.py (hunk 2 above)
Comment (GeorgeJJW) [reply picfixPy/sharpen.py:86]: Oh nevermind, I saw the `(intensity + 1) * input_img`, which expands and will include adding back the original image. Very nice!

**Review by GeorgeJJW — APPROVED** [review]
Good job implementing this function! I have to admit this is one of the harder ones out of the three. Feel free to issue another PR once you finish calibrating your test suite.

File: picfixPy/sharpen.py (hunk 1 above)
Comment (mani-kohli) [reply picfixPy/sharpen.py:62]: Thanks

**Pull request merged**